docs/data/material/experimental-api/css-variables/css-variables.md

```diff
@@ -0,0 +1,128 @@
+
+```jsx
+const Example() {
+  const theme = useTheme();
+  return <div style={{ color: theme.vars.palette.success.main }}>Success div</div>
+}
+```
+
+{{"demo": "SuccessDivCSSVariable.js", "hideToolbar": true }}
+
+```html
+<div style="color: var(--md-palette-success-main);">...</div>
+```
+
+Additionally, by clicking on the CSS variable in the browser's dev tools, you'll be taken to the theme object where you'll see its direct hex value again.
+We believe this to be a significant improvement in developer experience when debugging styles for any component.
+
+### Creating a custom theme
+
+The example above is using the default Material UI theme.
+But, what if you wanted to create your own custom theme?
+Let's see how to do that using the `experimental_extendTheme` utility.
+
+To start, it is important to understand this new theme structure.
+A theme is a collection of color schemes, each one defined within a single theme object.
+You'll be able to create more than light and dark color schemes if you want but, for the sake of simplicity, let's use only these two as an example. Here's how you'd customize it:
+
+```jsx
+const theme = experimental_extendTheme({
+  colorSchemes: {
+    light: {
+      palette: {
+        primary: teal,
+        secondary: deepOrange,
+      },
+    },
+    dark: {
+      palette: {
+        primary: cyan,
+        secondary: orange,
+      },
+    },
+  },
+});
+```
+
+{{"demo": "CssVarsCustomTheme.js", "iframe": true }}
+
+To toggle between color schemes, you can use the `useColorScheme` hook.
+This will make the browser re-write the CSS variables, pointing them to the new values you've created.
+The components' classes won't change at all, though—they should still refer to the same variables as before.
+
+### Customizing components
+
+As of now, the `Button` component is the only one supporting CSS variables (note that it doesn't cause any breaking change).
+To customize it using CSS variables, you'll need to wrap your application with `Experimental_CssVarisProvider`.
+Play around with the demo below!
+We'd appreciate any feedback about this API, as it is still in development.
+
+{{"demo": "CssVariablesCustomization.js", "iframe": true }}
+
+If you are using `TypeScript` you should use module augmentaiton to update the `Theme` structure:
+
+```tsx
+import { Theme as MuiTheme } from '@mui/material/styles';
+
+declare module '@mui/material/styles' {
+  interface Theme {
+    vars: Omit<
+      MuiTheme,
+      'typography' | 'mixins' | 'breakpoints' | 'direction' | 'transitions'
+    >;
+  }
+}
+```
+
+> To help us bring this new API to more components faster, feel free to contribute by adding CSS variables support in more components. Make sure to check the [GitHub issue](https://github.com/mui/material-ui/issues/32049) that keeps track of our progress first, in case someone took the lead already on the component you were thinking of.
```

Comment:
Rephrasing. Also, should this be near the top of the doc? I don't have a strong opinion but I could see it going either way.

```suggestion
> If you want to see wider support for this API across Material UI's component library, please feel free to contribute to the ongoing development. Make sure to check the [GitHub issue](https://github.com/mui/material-ui/issues/32049) that keeps track of our progress, to see if anyone else is currently working on a component you're interested in.
```